pages/docs/tutorials/native/using-intellij-idea.md

```diff
@@ -1,50 +1,145 @@
-system the project was created on (Windows, Linux, macOS).
+![Run the application]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-3.png') }})
 
-![Project]({{ url_for('tutorial_img', filename='native/using-intellij-idea/IDE-1.png')}})
+IntelliJ IDEA opens the **Run** tab and shows the output:
+![Application output]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-4.png') }})
 
-To run the project, simply invoke the [Run command in the IDE](https://www.jetbrains.com/help/idea/running-applications.html), by using the corresponding shortcuts or from the Run menu
+## Update the application
 
-![Run]({{ url_for('tutorial_img', filename='native/using-intellij-idea/IDE-2.png')}})
+### Count characters in your name
+
+1. Open the file `main.kt` in `src/<your_app_name>Main/kotlin`.
+
+   The `src` directory contains Kotlin source files and resources. The file `main.kt` includes sample code that prints "Hello, Kotlin/Native!" using the [prinln](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/println.html) function.
+
+2. Change the code of `main()`:
+
+   * Use the [readLine](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/read-line.html) function to read the input value to the `name` variable.
+   * Check the input value with the [safe call operator](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls) `?.`. 
+   * Use a [string template](https://kotlinlang.org/docs/reference/basic-types.html#string-templates) to insert your name into the string by adding a dollar sign `$` and enclosing it in curly braces – `${name?.length}`. 
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+   
+   ```kotlin
+   fun main() {
+       println("Hello, enter your name:")
+   
+       var name = readLine()
+       println("Your name contains ${name?.length} letters")
+   }
+   ```
+   
+   </div>
+
+3. Save the changes and run the application.
+
+   IntelliJ IDEA opens the **Run** tab and shows the output.
+
+4. Enter your name and enjoy the result:
+
+   ![Application output]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-5.png') }})
+
+### Eliminate the whitespaces
+
+1. Open the file `main.kt` in `src/<your_app_name>Main/kotlin`.
+
+2. Add a whitespaces check:
+
+   * Use the [contains](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/contains.html) function to check is there whitespaces in the input name .
+   * Use the [replace](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/replace.html) function to eliminate the whitespaces in the name.
+   * Change read-only variable `var name` to `val name` since you assign a new name value without whitespaces.
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+   
+   ```kotlin
+   val name = readLine()
+   println("Your name contains ${name?.length} letters")
```

Comment:
If you decide to use previous advice with null-check. Then here you can show that after null check no need to use operators `?.`
```
name?.let {
   println("Your name contains ${name.length} letters")
   if (name.contains(" ")) {
       name = name.replace(" ", "")
   }
   
   println("Your name without whitespaces: $name")
}
```

Reply:
At the same time I don't see any need to write such code. I'd rather write
`println("Your name without whitespaces: ${name.replace(" ", "")}")` without any conditions and changing variables.